website/docs/plugins/submitit_launcher.md

```diff
@@ -39,33 +39,85 @@ class QueueType(Enum):
 
 @dataclass
 class SlurmQueueConf:
-    nodes: int = 1
+    # Params are used to configure sbatch, for more info check:
+    # https://github.com/facebookincubator/submitit/blob/master/submitit/slurm/slurm.py
+
+    # maximum time for the job in minutes
+    time: int = 60
+    # number of cpus to use for each task
+    cpus_per_task: int = 10
+    # number of gpus to use on each node
     gpus_per_node: int = 1
+    # number of tasks to spawn on each node
     ntasks_per_node: int = 1
+    # number of nodes to use for the job
```

Comment:
Curious about these. I never used them in submitit:

for multitasking, does submitit itself launches the exact same job multiple times on the node?
how can the user control the differences between the jobs?
for example, if each job need a different config overrides?

Hydra 1.0 adds the ability to control the job environment [variables from the config](https://hydra.cc/docs/next/configure_hydra/job#hydrajobenv_set) which can be handy in some of these cases, for example to map the DDP rank_id to the hydra.job.num.

Reply:
> for multitasking, does submitit itself launches the exact same job multiple times on the node?

No `ntasks_per_node` and `nodes` are features of slurm, so we just forward the parameter to sbatch.

But be careful, it does not spawn multiple jobs, it is still a *unique job* (with a unique job id), but split on `nodes` machines, and on each machine the same code runs in `ntasks_per_node` processes/tasks. Each of the process is given a local and a global task rank which is used for synchronization. Environment variables (and submitit helpers) give you access to this.

In practice it's used for multi-GPUs jobs since one process/task will be responsible for one GPU (hence `gpus_per_node` will probably evolve to `gpus_per_task`)

> how can the user control the differences between the jobs?

I think the last 2 paragraph above explains it. There is only one job, running on several processes, and synchronization is done through environment variables (set by slurm in this case) which can be accessed through `submit.helpers.JobEnvironment` to unify between the different clusters (if available...)